src/main/java/fi/fmi/avi/converter/tac/taf/TAFTACParserBase.java

```diff
@@ -278,7 +283,9 @@ public void setTACLexer(final AviMessageLexer lexer) {
                         + "TAFReference");
             }
             //Note: Status & issue time of the referred report cannot be parsed from TAC
-            builder.setReferredReport(refBuilder.build());
+            if (refBuilder.getValidityTime().isPresent()) {
+                builder.setReferredReportValidPeriod(refBuilder.getValidityTime().get());
+            }
```

Comment:
The `TAFReference` interface is also deprecated. This whole `setReferredReport()` method should be rewritten and renamed as `setReferredReportValidPeriod()`.